src/terminal.c

```diff
@@ -431,52 +431,56 @@ name_lost_cb (GDBusConnection *connection,
 /* Copied from libcaja/caja-program-choosing.c; Needed in case
  * we have no DESKTOP_STARTUP_ID (with its accompanying timestamp).
  */
-static Time
-slowly_and_stupidly_obtain_timestamp (Display *xdisplay)
+static guint32
+slowly_and_stupidly_obtain_timestamp (GdkDisplay *display)
 {
-	Window xwindow;
-	XEvent event;
-
-	{
-		XSetWindowAttributes attrs;
-		Atom atom_name;
-		Atom atom_type;
-		const char *name;
-
-		attrs.override_redirect = True;
-		attrs.event_mask = PropertyChangeMask | StructureNotifyMask;
-
-		xwindow =
-		    XCreateWindow (xdisplay,
-		                   RootWindow (xdisplay, 0),
-		                   -100, -100, 1, 1,
-		                   0,
-		                   CopyFromParent,
-		                   CopyFromParent,
-		                   (Visual *)CopyFromParent,
-		                   CWOverrideRedirect | CWEventMask,
-		                   &attrs);
-
-		atom_name = XInternAtom (xdisplay, "WM_NAME", TRUE);
-		g_assert (atom_name != None);
-		atom_type = XInternAtom (xdisplay, "STRING", TRUE);
-		g_assert (atom_type != None);
-
-		name = "Fake Window";
-		XChangeProperty (xdisplay,
-		                 xwindow, atom_name,
-		                 atom_type,
-		                 8, PropModeReplace, (unsigned char *)name, strlen (name));
-	}
-
-	XWindowEvent (xdisplay,
-	              xwindow,
-	              PropertyChangeMask,
-	              &event);
-
-	XDestroyWindow(xdisplay, xwindow);
-
-	return event.xproperty.time;
+    if (GDK_IS_X11_DISPLAY (display)) {
+        Display *xdisplay;
+        Window xwindow;
+        XEvent event;
+        XSetWindowAttributes attrs;
+        Atom atom_name;
+        Atom atom_type;
+        char *name;
+        xdisplay = GDK_DISPLAY_XDISPLAY (display);
+
+
+        attrs.override_redirect = True;
+        attrs.event_mask = PropertyChangeMask | StructureNotifyMask;
+
+        xwindow = XCreateWindow (xdisplay, RootWindow (xdisplay, 0),
+                -100, -100, 1, 1,
+                0,
+                CopyFromParent,
+                CopyFromParent,
+                CopyFromParent,
+                CWOverrideRedirect | CWEventMask,
+                &attrs);
+
+        atom_name = XInternAtom (xdisplay, "WM_NAME", TRUE);
+        g_assert (atom_name != None);
+
+        atom_type = XInternAtom (xdisplay, "STRING", TRUE);
+        g_assert (atom_type != None);
+
+        name = "Fake Window";
+        XChangeProperty (xdisplay, xwindow, atom_name, atom_type,
+                8, PropModeReplace,
+                (unsigned char *) name, strlen (name));
+
+        XWindowEvent (xdisplay, xwindow,
+                PropertyChangeMask,
+                &event);
+
+        XDestroyWindow(xdisplay, xwindow);
+
+        return event.xproperty.time;
+    } else {
+        guint32 now;
+
+        now = (guint32) (g_get_real_time () / G_USEC_PER_SEC);
```

Comment:
I think you should use the [monotonic clock](https://developer.gnome.org/glib/stable/glib-Date-and-Time-Functions.html#g-get-monotonic-time) instead of the [real clock](https://developer.gnome.org/glib/stable/glib-Date-and-Time-Functions.html#g-get-real-time).

<img width="1525" alt="Captura de Pantalla 2020-04-15 a les 21 15 08" src="https://user-images.githubusercontent.com/10171411/79378805-3d47f200-7f5e-11ea-84d0-21e9056d1fec.png">

You should also use milliseconds instead of seconds:

<img width="1521" alt="Captura de Pantalla 2020-04-15 a les 21 23 50" src="https://user-images.githubusercontent.com/10171411/79379683-8d738400-7f5f-11ea-92a6-567abd133350.png">